docs/reference/dlm/index.asciidoc

```diff
@@ -0,0 +1,59 @@
+[role="xpack"]
+[[data-lifecycle-management]]
+== {dlm-init}: Manage your data lifecycle
+
+preview::[]
+
+The goal of {dlm} ({dlm-init}) is to enable you to easily automate the management of your data streams
+according to your retention requirements without having to worry. For example, you could use {dlm-init} to:
+
+* Ensure that data indexed in a data stream will be kept at least for the retention time you defined.
+* Ensure that data older than the retention period will be deleted automatically by {es} at a later time.
+
+To achieve that, it supports:
+
+* <<index-rollover,Rollover>>: which chunks your incoming data in smaller pieces to facilitate better performance
+and backwards incompatible mapping changes.
+* Configurable retention: allows you to configure the time period for which your data is guaranteed to be stored.
+{es} is allowed at a later time to delete data older than this time period.
+
+[discrete]
+[[dlm-how-it-works]]
+=== How does {dlm-init} work?
+
+{dlm-init} runs in intervals configured by <<indices-dlm-poll-interval,`indices.dlm.poll_interval`>>. At every run,
+{dlm-init} goes over each data stream and performs the following steps:
+
+. Checks if the data stream is managed by {dlm-init}, meaning it has a data lifecycle configured. If the data stream is
+managed by {dlm-init}, then
+. it will rollover the write backing index of the data stream, given it fulfills the conditions defined by
+<<cluster-dlm-default-rollover,`cluster.dlm.default.rollover`>>,
+. it will apply retention to the remaining backing indices. This means that it will delete  the backing indices that
+have completed the retention period after they have been rolled over. The time after rollover is called `generation_time`
+and it ensures that all the data in this index are at least older than this time.
```

Comment:
We are mentioning already in a previous part of this page and at another spot that the data gets deleted "at a later time". However, since we are explaining here how things work, I elaborated a bit more on the rationale behind the generation time and deletion. Let me know what you think.

I am a bit concerned that I am leaning towards too much information here.....